src/interpreter/Interpreter.ts

```diff
@@ -90,43 +91,54 @@ export class Interpreter {
+      this.interpretAndCombineMultipleSchemas(schema.anyOf, model, schema, interpreterOptions);
+      this.interpretAndCombineSchema(schema.then, model, schema, interpreterOptions);
+      this.interpretAndCombineSchema(schema.else, model, schema, interpreterOptions);
 
       interpretNot(schema, model, this, interpreterOptions);
     }
   }
 
   /**
-   * Go through schema(s) and combine the interpreted models together.
+   * Go through a schema and combine the interpreted models together.
    * 
    * @param schema to go through
    * @param currentModel the current output
+   * @param rootSchema the root schema to use as original schema when merged
+   * @param options to control the interpret process
    */
-  combineSchemas(schema: (Schema | boolean) | (Schema | boolean)[] | undefined, currentModel: CommonModel, rootSchema: Schema, interpreterOptions: InterpreterOptions = Interpreter.defaultInterpreterOptions): void {
+  interpretAndCombineSchema(schema: (Schema | boolean) | undefined, currentModel: CommonModel, rootSchema: Schema, interpreterOptions: InterpreterOptions = Interpreter.defaultInterpreterOptions): void {
     if (typeof schema !== 'object') {return;}
+    interpreterOptions = {...interpreterOptions, splitModels: false};
+    const models = this.interpret(schema, interpreterOptions);
+    if (models.length > 0) {
+      CommonModel.mergeCommonModels(currentModel, models[0], rootSchema);
+    }
+  }
+
+  /**
+   * Go through multiple schemas and combine the interpreted models together.
+   * 
+   * @param schema to go through
+   * @param currentModel the current output
+   * @param rootSchema the root schema to use as original schema when merged
+   * @param options to control the interpret process
+   */
+  interpretAndCombineMultipleSchemas(schema: (Schema | boolean)[] | undefined, currentModel: CommonModel, rootSchema: Schema, interpreterOptions: InterpreterOptions = Interpreter.defaultInterpreterOptions): void {
     if (Array.isArray(schema)) {
       schema.forEach((forEachSchema) => {
-        this.combineSchemas(forEachSchema, currentModel, rootSchema, interpreterOptions);
+        this.interpretAndCombineSchema(forEachSchema, currentModel, rootSchema, interpreterOptions);
       });
-    } else {
-      interpreterOptions = {...interpreterOptions, splitModels: false};
-      const models = this.interpret(schema, interpreterOptions);
-      if (models.length > 0) {
-        CommonModel.mergeCommonModels(currentModel, models[0], rootSchema);
-      }
     }
   }
 
   /**
-  * This function splits up a model if needed and add the new model to the list of models.
+  * This function splits up a model if it is determined it should.
   * 
   * @param model check if it should be split up
-  * @param models which have already been split up
   */
-  private splitModels(model: CommonModel): CommonModel {
+  private tryAndSplitModels(model: CommonModel): CommonModel {
```

Comment:
```suggestion
  private trySplitModels(model: CommonModel): CommonModel {
```

Reply:
I don't know why, but this sounds a bit weird as well 🤔? Does this make more sense for you?

Reply:
My problem is `tryAnd...`, but `try` what? 😄

Reply:
"try split models" vs "try and split models". When I read the name of the function the latter sounds more natural vs the first one 🤔

Reply:
As you want :) I don't think that we should use `And` and `Or` in function and variable names.

Reply:
Yea true.

Reply:
Sure lets change it 😄

Reply:
Changed

Reply:
Ok, If you want, let's wait for last review from czlowiek488 :)